src/App.tsx

```diff
@@ -1,26 +1,148 @@
-import React from 'react';
-import logo from './logo.svg';
+import * as React from 'react';
 import './App.css';
 
 function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+    const brd = [
+        [
+            { key: 1, id: '' },
+            { key: 2, id: '' },
+            { key: 3, id: '' },
+        ],
+        [
+            { key: 4, id: '' },
+            { key: 5, id: '' },
+            { key: 6, id: '' },
+        ],
+        [
+            { key: 7, id: '' },
+            { key: 8, id: '' },
+            { key: 9, id: '' },
+        ],
+    ];
+    const [board, setBoard] = React.useState([...brd]);
+    const [player, setPlayer] = React.useState('X');
+
+    const generateBoard = () => {
+        return board.map((elem, index) => (
+            <div key={index} className="rows">
+                {elem.map((elem, idx) => (
+                    <div key={idx} className={`box ${elem.id && 'disabled'}`} id={`${elem.key}`} onClick={boxOnClick}>
+                        {elem.id}
+                    </div>
+                ))}
+            </div>
+        ));
+    };
+
+    const boxOnClick = (e: React.MouseEvent<HTMLDivElement>) => {
+        const id = e.currentTarget.id;
+        let newBoard = [...board];
```

Comment:
```suggestion
        const newBoard = [...board];
```

newBoard can be `const`. In javascript const's are a little weird - you can modify their props